### ISSUE TITLE
Matching rejected actions in custom middleware

### ISSUE THREAD

**Issue by tobias-tengler** [reduxjs/redux-toolkit#782]
We want to log **all** rejected actions using a custom middleware.
For the time being we extracted the correct type (we assume?) from your type definitions and wrote a (bad) type guard for it:
```
import { Action, PayloadAction, SerializedError } from '@reduxjs/toolkit';

type RejectedAction = PayloadAction<
    undefined,
    string,
    {
        arg: any;
        requestId: string;
        aborted: boolean;
        condition: boolean;
    },
    SerializedError
>;

function isRejectedAction(action: Action): action is RejectedAction {
    return action.type.endsWith('/rejected');
}

export const loggerMiddleware: AppMiddleware = () => (next) => (action) => {
    next(action);

    if (isRejectedAction(action)) {
        const logger = Services.loggerFactory(action.type);

        logger.logError(action.error.message);
    }
};
```

Is there a better way for matching rejected actions dispatched from createAsyncThunk at the moment?
I know that individual actions have a `match` type guard function, is there something similar for a scenario like this?

**Comment by phryneas** [comment]
Currently not, but [we are a discussing a solution](https://github.com/reduxjs/redux-toolkit/issues/771#issuecomment-718104841) that will probably be part of the next version of RTK

**Comment by markerikson** [comment]
Fixed by #807 .
